redfish-core/lib/fan.hpp

```diff
@@ -326,6 +327,11 @@ inline void getFanAsset(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
         {
             asyncResp->res.jsonValue["Model"] = *model;
         }
+        
+	if (sparePartNumber != nullptr)
```

Comment:
Upstream we have been leaving this off if empty, oh well. 
This is a commit that could go upstream, so when you upstream, checking for empty will probably be required.